### PR TITLE
fix this validator

### PR DESCRIPTION
**Pull request by zsimjee**

### CODE REVIEW

File: tests/unit_tests.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+def compare_error_spans_lists(span_list1, span_list2):
+    if len(span_list1) != len(span_list2):
+        return False
+    for i in range(len(span_list1)):
+        if not compare_error_spans(span_list1[i], span_list2[i]):
+            return False
+    return True
+
+test_filter_output_in_sentence()
```
Comment (dtam) [inline tests/unit_tests.py:71]: are these executions necessary?

File: tests/unit_tests.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+test_compute_filtered_output()
+test_compute_error_spans()
+test_find_competitor_matches()
```
Comment (zsimjee) [inline tests/unit_tests.py:74]:
```suggestion
```

File: validator/main.py [hunk 3]
```diff
@@ -243,6 +195,81 @@ def _inference_remote(self, model_input: Any) -> str:
...
+            return sentence
+    
+        filtered_text = sentence
+        
+        for competitor in competitors:
+            filtered_text = filtered_text.replace(competitor, "[COMPETITOR]")
+
+        return filtered_text
+            
+    def compute_error_spans(self, sentences: List[str], competitors_per_sentence: List[List[str]]) -> str:
```
Comment (dtam) [inline validator/main.py:253]:
```suggestion
    def compute_error_spans(self, sentences: List[str], competitors_per_sentence: List[List[str]]) -> List[ErrorSpan]:
```

**Comment by dtam** [comment]
tested locally and works.

**Pull request merged**